contracts/btc-staking/src/staking.rs

```diff
@@ -230,6 +230,18 @@ pub fn handle_active_delegation(
         ACTIVATED_HEIGHT.save(storage, &(height + 1))?; // Active from the next block onwards
     }
 
+    // Index the delegation by its end height
+    BTC_DELEGATION_EXPIRY_INDEX.update(
```

Comment:
This may be inefficient, as at least in principle there could be a good number of delegations by end height.

I think the best way to do this would be to turn the `BTC_DELEGATIONS` `Map` into an `IndexedMap`, and add a multi-index of staking tx hashes per end height.

There are some examples of `MultiIndex` usage in the contracts, but doing this change is involved. Perhaps put a TODO / FIXME in here for now?